### PR TITLE
feat(slack): add guest user import support

### PR DESCRIPTION
**Pull request by fmartingr**
## Summary

Adds Slack guest user support to `mmetl transform slack`. Slack guests (`is_restricted` multi-channel guests and `is_ultra_restricted` single-channel guests) were previously imported as full Mattermost members, which is a security/compliance concern requiring manual cleanup after every migration.

Fixes [MM-69710](https://mattermost.atlassian.net/browse/MM-69710).

## What's included

- **Guest detection**: `SlackUser.IsGuest()` reads `is_restricted`/`is_ultra_restricted`; `IntermediateUser.IsGuest` carries it through the shared intermediate representation.
- **`--guest-handling` flag** on `transform slack` (`guest` default / `user` / `skip`), matching the pattern established by the in-flight RocketChat importer (#96):
  - `guest` — migrate as Mattermost guests (`system_guest`/`team_guest`/`channel_guest`). Requires Guest Accounts to be licensed and enabled on the destination server.
  - `user` — migrate as regular members (works without guest licensing).
  - `skip` — drop guest users entirely, along with their memberships and authored posts/reactions.
- **Shared package changes** (`services/intermediate`): `Exporter.EmitGuestRoles` and a `GetImportLineFromUser(user, team, emitGuestRoles)` signature shared with #96, so both providers follow the same contract instead of diverging. Includes a safety fallback: a guest with zero channel memberships is exported as a regular member instead of producing an invalid bulk-import line (Mattermost requires guest role consistency across system/team/channel scope, or the whole import aborts).
- **DM/group channels**: guests are marked `scheme_guest` (not `scheme_user`) in direct/group channel participant lists too, via the same `isEffectiveGuest` check used for the user-level roles, so a guest's status is consistent across every line that references them.
- **Docs**: README section on Slack guest handling (including the Guest Accounts licensing caveat) and regenerated `docs/cli/mmetl_transform_slack.md`.

## Test plan

- [x] Unit tests validate role output against Mattermost's real `imports.ValidateUserImportData` validator for multi-channel guest, single-channel guest, `user`-mode, and the zero-membership edge case.
- [x] Unit tests cover `guest-handling=guest|user|skip` in `TransformUsers`.
- [x] Unit tests cover `SchemeGuest`/`SchemeUser` on direct-channel participants.
- [x] Docker-backed e2e test (`TestTransformSlackE2EGuestImport`) imports into a real Mattermost server via `mmctl` and verifies `IsGuest()`/`SchemeGuest` at team, channel, and DM level, including that a single-channel guest is correctly scoped out of a channel it never belonged to, and that a guest in a DM is `scheme_guest` there too.
- [x] `make check-style` and `make test` (full suite, including all Docker e2e tests) pass.

### CODE REVIEW

File: services/intermediate/export.go [hunk 1]
```diff
@@ -339,7 +390,10 @@ func (e *Exporter) ExportUsers(writer io.Writer, botOwner string) error {
 
 	// Write regular users first (bot owner must exist before bots)
 	for _, user := range users {
-		line := GetImportLineFromUser(user, e.TeamName)
+		if e.EmitGuestRoles && user.IsGuest && len(user.Memberships) == 0 {
+			e.Logger.Warnf("Guest user %s has no channel memberships; importing as a regular member instead, since Mattermost requires guests to have at least one channel", user.Username)
```
Comment (vish9812) [inline services/intermediate/export.go:419]: a `guest` with no channels becomes a full team member, who can then browse/join every public channel — the opposite of what `guest mode` is for. Keeping them a guest isn't possible (all no-channel guest shapes fail `isValidGuestRoles`), so the choice is skip vs. promote. Suggest defaulting to skip and gating member-promotion behind an explicit flag/value rather than making it the silent default.

File: services/slack/intermediate.go [hunk 2]
```diff
@@ -1046,6 +1103,11 @@ func (t *Transformer) Transform(slackExport *SlackExport, attachmentsDir string,
 	t.ComputeChannelPostStats()
 	t.applyChannelStatsToMemberships()
 
+	if t.droppedPostRefs > 0 || t.droppedMembershipRefs > 0 {
+		t.Logger.Infof("Dropped %d posts and %d channel/DM memberships referencing skipped users",
```
Comment (vish9812) [inline services/slack/intermediate.go]: Undercount here: when `skip` drops a `guest` who authored a thread root, replies from non-skipped users hit the "couldn't find rootPost" branch in `AddPostToThreads` and are dropped without incrementing `droppedPostRefs`. So legitimate (non-guest) content is lost and this summary reads as more complete than it is. At minimum count those; ideally reparent or surface them.

File: commands/check.go [hunk 3]
```diff
@@ -90,7 +90,7 @@ func checkSlackCmdF(cmd *cobra.Command, args []string) error {
 		return err
 	}
 
-	err = slackTransformer.Transform(slackExport, "", true, true, false, skipEmptyEmails, defaultEmailDomain)
+	err = slackTransformer.Transform(slackExport, "", true, true, false, skipEmptyEmails, defaultEmailDomain, slack.GuestHandlingGuest)
```
Comment (vish9812) [inline commands/check.go]: `check` hardcodes guest mode (no --guest-handling flag) and never calls `Export`, so the downgrade warning — which lives in `ExportUsers` — never fires here. Combined with logs going to `check-slack.log` rather than `stdout`, operators won't see any of this at check time even though check is the natural place to preview it. Suggest adding the flag to check and surfacing the guest decisions to stdout/summary.

File: services/intermediate/export.go [hunk 4]
```diff
@@ -304,11 +348,18 @@ func (e *Exporter) ExportChannels(channels []*IntermediateChannel, writer io.Wri
 
 // ExportDirectChannels is valid for group or direct channels, as they export with members.
 func (e *Exporter) ExportDirectChannels(channels []*IntermediateChannel, writer io.Writer) error {
+	guestUsernames := make(map[string]bool)
```
Comment (vish9812) [inline services/intermediate/export.go]: Nit: this map is rebuilt on each ExportDirectChannels call (group + direct = twice). Could hoist onto the Exporter.

**Review by vish9812 — CHANGES_REQUESTED** [review]
Great work here.

One design ask and a couple of smaller fixes:

The `channel-less-guest` fallback defaults to the **most-permissive** outcome. The fallback itself is necessary — a `system_guest` with no channels fails `isValidGuestRoles` and aborts the whole import. But downgrading to a `member` puts the account on the team, and team members can _browse/join every public channel_ — so a user with zero Slack channel access ends up with access to all of them, _silently_, in the mode meant to contain guests. A channel-less user can't be kept as a guest in any form, so the only real question is what to do with them:

- skip (default) — drop the account.
- promote to member — behind an explicit flag/value, not the default. Preserves the account, but grants full public-channel access now and stays escalatable.

Suggestion: default channel-less guests to `skip` and require an **explicit opt-in** to import them as members.

**Review by vish9812 — APPROVED** [review]
